File: src/TaskManager.Domain/Entities/project.cs
namespace TaskManager.Domain.Entities;

public class Project
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Relación 1 → N: un proyecto tiene muchas tareas
    public ICollection<TaskItem> Tasks { get; set; } = new List<TaskItem>();
}

File: client/src/types/project.ts
export interface Project {
    id: number;
    name: string;
    description: string | null;
    createdAt: string;
    taskCount: number;
  }
  
  export interface CreateProjectInput {
    name: string;
    description?: string | null;
  }
  
  export interface UpdateProjectInput {
    name: string;
    description?: string | null;
  }